test-utils/src/lib.rs

```diff
@@ -88,14 +93,18 @@ impl BlockBuilder {
             self.transactions
         };
 
-        let merkle_root =
-            bitcoin::util::hash::bitcoin_merkle_root(txdata.iter().map(|tx| tx.txid().as_hash()))
-                .unwrap();
-        let merkle_root = TxMerkleNode::from_hash(merkle_root);
+        let merkle_root = bitcoin::merkle_tree::calculate_root(
+            txdata
+                .iter()
+                .map(|tx| *tx.compute_txid().as_raw_hash())
+                .clone(),
+        )
+        .unwrap();
+        let merkle_root = TxMerkleNode::from_raw_hash(merkle_root);
 
         let header = match self.prev_header {
-            Some(prev_header) => header(&prev_header, merkle_root),
             None => genesis(merkle_root),
+            Some(prev_header) => header(&prev_header, merkle_root),
```

Comment:
Out of curiosity, why did this line move down?

Reply:
I thought it's easier to read when the first line is the logical start of the blockchain -- genesis, and all the rest is a header calculation based on a prev_header.